prompting/apparmor/modeset.go

```diff
@@ -0,0 +1,63 @@
+package apparmor
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ModeSet is a bit mask of apparmor "modes".
+type ModeSet uint32
+
+const (
+	APPARMOR_MODESET_AUDIT ModeSet = 1 << iota
+	APPARMOR_MODESET_ALLOWED
+	APPARMOR_MODESET_ENFORCE
+	APPARMOR_MODESET_HINT
+	APPARMOR_MODESET_STATUS
+	APPARMOR_MODESET_ERROR
+	APPARMOR_MODESET_KILL
+	APPARMOR_MODESET_USER
+)
+
+const modeSetMask = APPARMOR_MODESET_AUDIT | APPARMOR_MODESET_ALLOWED | APPARMOR_MODESET_ENFORCE | APPARMOR_MODESET_HINT | APPARMOR_MODESET_STATUS | APPARMOR_MODESET_ERROR | APPARMOR_MODESET_KILL | APPARMOR_MODESET_USER
+
+// String returns readable representation of the mode set value.
+func (m ModeSet) String() string {
+	frags := make([]string, 0, 8)
+	if m&APPARMOR_MODESET_AUDIT != 0 {
+		frags = append(frags, "audit")
+	}
+	if m&APPARMOR_MODESET_ALLOWED != 0 {
+		frags = append(frags, "allowed")
+	}
+	if m&APPARMOR_MODESET_ENFORCE != 0 {
+		frags = append(frags, "enforce")
+	}
+	if m&APPARMOR_MODESET_HINT != 0 {
+		frags = append(frags, "hint")
+	}
+	if m&APPARMOR_MODESET_STATUS != 0 {
+		frags = append(frags, "status")
+	}
+	if m&APPARMOR_MODESET_ERROR != 0 {
+		frags = append(frags, "error")
+	}
+	if m&APPARMOR_MODESET_KILL != 0 {
+		frags = append(frags, "kill")
+	}
+	if m&APPARMOR_MODESET_USER != 0 {
+		frags = append(frags, "user")
+	}
+	if residue := m &^ modeSetMask; residue != 0 {
```

Comment:
(nitpick) I wonder if `residue` is the best term here, it describes the unaccounted bits andif there areany this is no longer valid. So maybe `unaccounted`, `unknown`, `extra`, `unexpected` or something? But not a native speaker and residue maybe perfectly fine :) [also this may come from the C  libapparmor)

Reply:
I like `unaccounted` as well. I checked, there is no reference to `residue` in libapparmor, nor in `/usr/include/{sys,linux}/apparmor.h`.